src/settings/components/account-settings/settings-sent-messages.tsx

```diff
@@ -0,0 +1,284 @@
+/*
+ * SPDX-FileCopyrightText: 2021 Zextras <https://www.zextras.com>
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+
+import React, { useMemo, useCallback, ReactElement, useState, useEffect } from 'react';
+import {
+	Container,
+	Text,
+	Padding,
+	Input,
+	Row,
+	Select,
+	Checkbox,
+	Dropdown,
+	Icon
+} from '@zextras/carbonio-design-system';
+import { TFunction } from 'i18next';
+import { filter } from 'lodash';
+import { IdentityProps } from '../../../../types';
+
+type SettingsSentMessagesProps = {
+	t: TFunction;
+	items: IdentityProps;
+	isExternalAccount: boolean;
+	setMods: (mods: { [key: string]: unknown }) => void;
+	updateIdentities: (id: string, key: string, pref: string) => void;
+};
+
+const SettingsSentMessages = ({
+	t,
+	items,
+	isExternalAccount,
+	setMods,
+	updateIdentities
+}: SettingsSentMessagesProps): ReactElement => {
+	const title = useMemo(() => t('label.settings_sent_messages', 'Settings for Sent Messages'), [t]);
+
+	const [replyToEnabled, setReplyToEnabled] = useState(items.replyToEnabled === 'TRUE');
+
+	useEffect(() => {
+		setReplyToEnabled(items.replyToEnabled === 'TRUE');
+	}, [items.replyToEnabled]);
+
+	const checkboxOnClick = useCallback(() => {
+		setReplyToEnabled(!replyToEnabled);
+		if (!replyToEnabled === (items.replyToEnabled === 'TRUE')) {
+			setMods({});
+		} else {
+			updateIdentities(
+				items.identityId,
+				'zimbraPrefReplyToEnabled',
+				replyToEnabled ? 'FALSE' : 'TRUE'
+			);
+		}
+	}, [items.replyToEnabled, replyToEnabled, setMods, updateIdentities, items.identityId]);
+
+	const [dropdownOpen, setDropdownOpen] = useState(false);
+	const [fromDisplayValue, setFromDisplayValue] = useState(items.fromDisplay || ' ');
+	useEffect(() => {
+		setFromDisplayValue(items.fromDisplay || ' ');
+	}, [items.fromDisplay]);
+
+	const fromDisplayLabel = useMemo(
+		() => (fromDisplayValue === (undefined || '') ? t('label.from_name', 'From: "Name"') : ''),
+		[t, fromDisplayValue]
+	);
+	const onChangeFromDisplayValue = (
+		ev: MouseEvent & {
+			target: HTMLButtonElement;
+		}
+	): void => {
+		setFromDisplayValue(ev.target.value);
+		if (ev.target.value === items.fromDisplay) {
+			setMods({});
+		} else {
+			updateIdentities(items.identityId, 'zimbraPrefFromDisplay', ev.target.value);
+		}
+	};
+
+	const fromAddressLabel = useMemo(() => t('label.address', 'Address'), [t]);
+	const fromAddressArray = useMemo(
+		() => [{ value: items.fromAddress, label: items.fromAddress }],
+		[items]
+	);
+	const [fromAddress, setFromAddress] = useState(
+		filter(fromAddressArray, (item) => item.value === items.fromAddress)[0]
```

Comment:
find can work better here

Reply:
yeah, especially when picking out the first element of the result.